provider/service.py

```diff
@@ -109,7 +104,20 @@ def run(self):
                                 elif triggerIsAssignedToMe:
                                     logging.info('[{}] Found a change to an existing trigger'.format(change["id"]))
 
-                                    if existingConsumer.desiredState() == Consumer.State.Disabled and self.__isTriggerDocActive(document):
+                                    if existingConsumer.desiredState() == Consumer.State.Dead and self.__isTriggerDocActive(document):
```

Comment:
For background info, there is a nanny runs periodically that delete consumers that are in a dead state. Since the consumer is already in a dead state waiting for the nanny to close the process, a quick recreate of the consumer -- within a 2 second time period -- will be ignored.